### DISCUSSION TITLE
How to default tests to run on one worker and then opt in case by case

### DISCUSSION THREAD

**Discussion by pauldambra** [pytest-dev/pytest-xdist#942] [General]
We have a large test suite and it won't be straight-forward to convert to running in parallel but it's getting very slow 😅 

Since conversion won't be straight-forward I'd like to set all tests to run on one worker and then opt files/classes in to running on multiple workers.

Is this possible?

**Comment by RonnyPfannschmidt** [comment]
Currently there's no affinity mechanism

**Reply by pauldambra** [reply]
Thanks

I was wondering about setting a default group and then opting in using further groups. I'm guessing that's not possible?

**Reply by pauldambra** [reply]
It looked to me like any code I added to the `pytest_collection_modifyitems` hook wasn't affecting xdist. 

I'd like to add a marker to any test that doesn't already have one so I can default all tests into one group. Is that right? Is there an equivalent hook in xdist?

**Reply by nicoddemus** [reply]
`pytest_collection_modifyitems` works with xdist, the only catch is that it only runs on the workers, not on `master`.

**Reply by pauldambra** [reply]
Ah, that should be fine for me... I was splitting my attention and probably did something silly. I'll check again. Thank you 🙏

**Reply by RonnyPfannschmidt** [reply]
Workers don't decide about scheduling,so the hook can't help your case

**Reply by pauldambra** [reply]
My current thought is that I set xdist_group to a default value if it isn't present. Then all tests would run on one worker until we opt then into a different group
